core/src/main/java/com/linecorp/armeria/common/DefaultConcurrentAttributes.java

```diff
@@ -379,6 +389,8 @@ private enum SetAttrMode {
         @Nullable
         private DefaultAttribute<?> next;
 
+        private final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
How about making `DefaultAttribute` extend `ReentrantLock` so that we can reduce the number of creating instances by one?
`static final class DefaultAttribute<T> extends ReentrantLock implements Entry<AttributeKey<T>, T> {...}`

Reply:
apply it, removed lock and unlock api that i created!

Reply:
I was faced with a situation where my tests were running infinitely.
I did my testing by reverting line by line.
As result, This recommendation seems to be causing the problem.(DefaultAttribute extend ReentrantLock)
Could you please review to make sure I'm not implementing your suggestion incorrectly?🙇🏻‍♂️ @minwoox